src/python/pants/backend/pants_info/explain_options.py

```diff
@@ -190,3 +220,32 @@ def console_output(self, targets):
                         inner_map["history"] = history_list
         if self.is_json():
             yield json.dumps(output_map, indent=2, sort_keys=True, cls=CoercingOptionEncoder)
+
+
+@goal_rule
+def explain_options(
+    explain_options_options: ExplainOptionsOptions,
```

Comment:
Maybe call it `ExplainOptionsSubsystem` or `GoalSubsystem`. It diverges from convention, but maybe worth it.